### PR TITLE
Update tox4j dependencies

### PR DESCRIPTION
**Pull request by roihershberg**
Update `protobuf` and `libvpx` while modifying `libvpx.patch` and compiling with clang instead of made up gcc links that point to clang.

### CODE REVIEW

**Comment by roihershberg** [comment]
Those things are very hard for me. Absolutely no experience and I hardly understand all of this stuff. The build passes but I have no idea if `libvpx` works on Android with these changes.

**Comment by robinlinden** [comment]
I'm also not 100% sure what effect this will have. I'll build this locally to run some tests on real Android devices. Looks good though!

**Comment by robinlinden** [comment]
While removing the patch would be amazing, I don't think switching to generic-gnu for the target in vpx is a good idea. Audio/video is the most demanding part of aTox, so we should be doing what we can to keep it somewhat reasonable. :x

Any chance you can look into keeping the correct architecture-specific optimizations?

File: scripts/build-aarch64-linux-android [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/make -f
 
 TARGET := aarch64-linux-android
+BASE_TARGET := $(TARGET)
```
Comment (robinlinden) [inline scripts/build-aarch64-linux-android:4]: I guess this serves the same purpose as the `VPX_ARCH` used previously? You can delete `VPX_ARCH` now that it's unused. :P

**Comment by robinlinden** [comment]
How much have you tested this? I guess we still need to set up a proper Android test to make sure it all works well?

File: scripts/build-aarch64-linux-android (hunk 1 above)
Comment (roihershberg) [reply scripts/build-aarch64-linux-android:4]: Right

**Comment by roihershberg** [comment]
I haven't tested it because I don't know how to as video is not yet implemented. A proper test should indeed be implemented.
Maybe we can take those tests you mentioned and add them. I would need to learn Scala and androidTest before.

**Comment by robinlinden** [comment]
We probably don't want to add any Scala since this project is 100% Kotlin right now, but since our build scripts are pretty identical, I figured we could do any of
- use https://github.com/scala-android/sbt-android to test these changes directly in the tox4j repo
- set up a temporary hack branch that just runs the tests as-is, but in our repo (Scala and all)
- take inspiration from those tests when setting up more real permanent tests in this repo, but written in Kotlin
- build vpx .so files with the new patches and test in a different project that has a/v set up

**Comment by roihershberg** [comment]
I prefer option 3. Taking inspiration and making our own tests.

**Comment by roihershberg** [comment]
But for the sake of merging this PR we can do one of the first 2 options you suggested. After that we can make proper tests in general to more than av.

**Comment by robinlinden** [comment]
This PR looks better than what we had, so if testing is too difficult, an option could be to merge it, test later, and revert it if it doesn't work. :x Not patching the NDK is a massive win imo.

There's some form of video support in [Antox][antox-video], so I guess building Antox against tox4j built with these scripts should work too. That might be the easiest option for now?

[antox-video]: https://github.com/Antox/Antox/blob/master/app/src/main/scala/chat/tox/antox/av/VideoDisplay.scala

**Comment by roihershberg** [comment]
Yeah we can try with Antox. And if something in the process doesn't work then we can just merge it and test later like you said.

**Comment by roihershberg** [comment]
I tested Antox with the tox4j artifacts of this PR and it works. But it works more slowly and it has some difficulties. Maybe there's more data to send compared to the old version. I assume that `libvpx` has changed a lot since but maybe it can also be optimized. The video from my PC camera with the low resolution (maybe 240p) shows almost instantly in my phone but the video from my phone takes a long time until it shows on my PC and it has some glitches due to the delays.

**Comment by robinlinden** [comment]
Sounds like a good start then! We can do proper performance testing once we've set up video support in aTox. :P

**Review by robinlinden — APPROVED** [review]

**Pull request merged**